inlong-sdk/dataproxy-sdk/src/main/java/org/apache/inlong/sdk/dataproxy/utils/HashRing.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.sdk.dataproxy.utils;
+
+import org.apache.inlong.sdk.dataproxy.config.HostInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.TreeMap;
+import java.util.SortedMap;
+import java.util.stream.Collectors;
+
+public class HashRing {
+    private int virtualNode = 1000;
+    private TreeMap<String, HostInfo> virtualNode2RealNode = new TreeMap<>();
+    private List<HostInfo> nodeList = new ArrayList<>();
+    private static final HashRing instance = new HashRing();
+    private static final Logger LOGGER = LoggerFactory.getLogger(HashRing.class);
+
+    public static HashRing getInstance() {
+        return instance;
+    }
+
+    public TreeMap<String, HostInfo> getVirtualNode2RealNode() {
+        return virtualNode2RealNode;
+    }
+
+    public String node2VirtualNode(HostInfo node, int index) {
+        return  "virtual&&" + index + "&&" + node.toString();
+    }
+
+    public void initHashRing(List<HostInfo> ipList) {
```

Comment:
since the instance has been constructed at the very beginning, why not directly init it at the constructor?
if multi thread call initHashRing, what will happend?